### PR TITLE
Frontend: Adapt to new fee model

### LINKED ISSUE #617: Adapt frontend for fee model changes
Related to #607 

- [x] Needs design (create separate issue)
- [ ] Remove displaying the amount expected to be paid.
- [ ] Don't send ETH with request.
- [ ] Display fee amount expected to be paid @taleldayekh to split into separate issue.
- [ ] Add button to send all token which will automatically calculate the value that can be sent @taleldayekh to split into separate issue.

### PR DESCRIPTION
**Pull request by manuelwedler**
Closes #617 

This updates all modules to the new fee model. The contract abis were updated, but this also breaks the frontend on main branch (once this is merged), because there is no deployment of the new contracts. As our documentation provides a deterministic local test setup, I added a configuration for it in order to not completely break main branch. There is also some refactoring on the token and chain selectors in order to avoid some boilerplate code. See the commit messages for details.

### CODE REVIEW

File: frontend/src/composables/useChainSelection.ts
Comment (manuelwedler) [reply frontend/src/composables/useChainSelection.ts:34]: This means we don't filter the selected source chain from the target chains in development, because there will only be one chain. Maybe the expression is not very expressive?

Comment (manuelwedler) [reply frontend/src/composables/useChainSelection.ts:34]: But I  think the expression was wrong anyway. And the chain id should be compared not the object.

File: frontend/tests/unit/composables/useRequestFee.spec.ts
Comment (manuelwedler) [reply frontend/tests/unit/composables/useRequestFee.spec.ts:22]: Nice! I will update all files to this scheme. I think it's much better. It's always good to get rid of the `eslint-disable` comments.

File: frontend/tests/unit/composables/useRequestFee.spec.ts
Comment (manuelwedler) [reply frontend/tests/unit/composables/useRequestFee.spec.ts:30]: Ok makes sense. So I will only define the values locally in the function if the values are actually tested.

File: frontend/tests/unit/composables/useRequestFee.spec.ts
Comment (manuelwedler) [reply frontend/tests/unit/composables/useRequestFee.spec.ts:113]: I see, we want to test against the actual string here. Makes totally sense to change it.

File: frontend/tests/unit/composables/useRequestFee.spec.ts
Comment (weilbith) [reply frontend/tests/unit/composables/useRequestFee.spec.ts]: Yeah kinda. It won't be always possible and sometimes it makes sense to have exceptions because it improves the readability or so. But I think in general it is a good approach to keep test cases easy to read and understandable.

File: frontend/src/composables/useChainSelection.ts
Comment (weilbith) [reply frontend/src/composables/useChainSelection.ts]: Haha okay.
But please just for the record: why do we need this filter adaption for `development` anyway?

Comment (weilbith) [reply frontend/src/composables/useChainSelection.ts]: 👌🏾

File: frontend/tests/unit/composables/useRequestFee.spec.ts
Comment (weilbith) [reply frontend/tests/unit/composables/useRequestFee.spec.ts]: 👌🏾

File: frontend/tests/unit/composables/useRequestFee.spec.ts
Comment (weilbith) [reply frontend/tests/unit/composables/useRequestFee.spec.ts]: 👌🏾

File: frontend/tests/unit/composables/useRequestFee.spec.ts
Comment (weilbith) [reply frontend/tests/unit/composables/useRequestFee.spec.ts]: 👌🏾

File: frontend/src/composables/useChainSelection.ts
Comment (manuelwedler) [reply frontend/src/composables/useChainSelection.ts]: In the local test deployment, we do the transfers to the same target as the source chain (Ganache). Without the check, the target chain would get filtered and the selector would be empty.
This reasoning is also in the commit message.

Comment (weilbith) [reply frontend/src/composables/useChainSelection.ts]: Oh okay. Sorry that I missed that. Especially since I put such information in commit messages too. 🙈

**Comment by weilbith** [comment]
Please do not merge before #712 and #735 so we can a last testnet release before this breaking change gets in.

**Comment by weilbith** [comment]
All PRs that should be merged before are finished now. The last testnet release for these contracts is building. Let's rebase these changes and resolve the conflicts.

**Comment by weilbith** [comment]
Alright, all conflicts are resolved and the branch is clean. Though I would still always do a final manual test. Unfortunately it is quite late already and I've never used the local setup so far. So I'll do it tomorrow in the office. :v:

**Review by weilbith — APPROVED** [review]

**Comment by weilbith** [comment]
Alright, so first of all, there are some issues that got introduced with the rebasing on a mass of other features. And second, because one of these new features was expired request handling, we have more breaking changes to handle. It isn't nice, but we will fix that issue here in this PR too.

**Comment by weilbith** [comment]
Doing a final manual test. The positive case worked fine already. 
Once this PR got merged, we will check with the agents and do a new `testnet.` release.

**Comment by weilbith** [comment]
I forgot something... 😑

**Pull request merged**
